### PR TITLE
feat: implement repository setup for monorepo typescript with examples and packages

### PR DESCRIPTION
**Pull request by fbozic**
- setup repository level config for typescript monorepo
- setup packages/calimero-sdk
- setup examples/hello-world

### CODE REVIEW

File: packages/calimero-sdk/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+    "compilerOptions": {
+        "esModuleInterop": true,
+        "lib": [
+            "es2015",
+            "esnext",
+            "dom"
+        ],
+        "module": "commonjs",
```
Comment (MatejVukosav) [inline packages/calimero-sdk/tsconfig.json]: Use `ESNext` instead of commonjs.

File: packages/calimero-sdk/package.json [hunk 2]
```diff
@@ -0,0 +1,59 @@
+{
+    "name": "calimero-p2p-sdk",
+    "version": "0.0.1",
+    "description": "Javascript library to interact with Calimero node",
+    "main": "lib/index.js",
```
Comment (MatejVukosav) [inline packages/calimero-sdk/package.json:6]: add  ` "type": "module",` for esnext

File: packages/calimero-sdk/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        "target": "es2015",
```
Comment (MatejVukosav) [inline packages/calimero-sdk/tsconfig.json]: ESNext

File: packages/calimero-sdk/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        "moduleResolution": "node",
+        "alwaysStrict": true,
+        "outDir": "./lib",
+        "declaration": true,
+        "preserveSymlinks": true,
+        "preserveWatchOutput": true,
+        "pretty": false,
+        "forceConsistentCasingInFileNames": true,
+        "noFallthroughCasesInSwitch": true,
+        "noImplicitAny": false,
```
Comment (MatejVukosav) [inline packages/calimero-sdk/tsconfig.json]: true

File: examples/only-peers-simple/package.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  "dependencies": {
+    "@testing-library/jest-dom": "^5.16.5",
+    "@testing-library/react": "^13.4.0",
+    "@testing-library/user-event": "^13.5.0",
+    "calimero-p2p-sdk": "file:../../packages/calimero-sdk",
+    "path-browserify": "^1.0.1",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0",
+    "react-router-dom": "^6.8.2",
+    "react-scripts": "5.0.1",
```
Comment (xilosada) [inline examples/only-peers-simple/package.json]:
```suggestion
```

File: examples/only-peers-simple/package.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    "web-vitals": "^2.1.4"
```
Comment (xilosada) [inline examples/only-peers-simple/package.json]:
```suggestion
```

File: examples/only-peers-simple/package.json [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    "dev": "vite",
+    "build": "vite build",
+    "preview": "vite preview"
+  },
+  "dependencies": {
+    "@testing-library/jest-dom": "^5.16.5",
+    "@testing-library/react": "^13.4.0",
+    "@testing-library/user-event": "^13.5.0",
+    "calimero-p2p-sdk": "file:../../packages/calimero-sdk",
+    "path-browserify": "^1.0.1",
```
Comment (xilosada) [inline examples/only-peers-simple/package.json]: Do we need this?

File: examples/only-peers-simple/vite.config.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import { defineConfig } from 'vite';
+import react from '@vitejs/plugin-react';
+import path from 'path';
+
+export default defineConfig({
+  plugins: [react()],
+  resolve: {
+    fallback: {
+      fs: false,
+      path: path.resolve(__dirname, 'node_modules/path-browserify'),
```
Comment (xilosada) [inline examples/only-peers-simple/vite.config.js]: Check the node-ui config. There is a lib to patch all polyfills

File: examples/only-peers-simple/package.json (hunk 7 above)
Comment (fbozic) [reply examples/only-peers-simple/package.json]: looks like we don't, removed

**Pull request merged**